phys2bids/phys2bids.py

```diff
@@ -137,39 +135,54 @@ def use_heuristic(heur_file, sub, ses, filename, outdir, record_label=''):
     -------
     heurpath: str or path
         Returned fullpath to tsv.gz new file (post BIDS formatting).
+
+    Raises
+    ------
+    KeyError
+        if `bids_keys['task']` is empty
     """
+    utils.check_file_exists(heur_file)
 
-    if sub[:4] != 'sub-':
-        name = f'sub-{sub}'
-    else:
-        name = sub
+    # Initialise a dictionary of bids_keys that has already "recording"
+    bids_keys = {'sub': '', 'ses': '', 'task': '', 'acq': '', 'ce': '',
+                 'dir': '', 'rec': '', 'run': '', 'recording': record_label}
 
-    fldr = os.path.join(outdir, name)
+    # Start filling bids_keys dictionary and path with subject and session
+    if sub[:4] == 'sub-':
+        bids_keys['sub'] = sub[4:]
+        fldr = os.path.join(outdir, sub)
+    else:
+        bids_keys['sub'] = sub
+        fldr = os.path.join(outdir, f'sub-{sub}')
 
     if ses:
-        if ses[:4] != 'ses-':
-            ses = f'ses-{ses}'
-
-        fldr = os.path.join(fldr, ses)
-        name = f'{name}_{ses}'
+        if ses[:4] == 'ses-':
+            bids_keys['ses'] = ses[4:]
+            fldr = os.path.join(fldr, ses)
+        else:
+            bids_keys['ses'] = ses
+            fldr = os.path.join(fldr, f'ses-{ses}')
 
-    fldr = os.path.join(fldr, 'func')
-    utils.path_exists_or_make_it(fldr)
+    # Load heuristic and use it to fill dictionary
+    heur = utils.load_heuristic(heur_file)
+    bids_keys = heur.heur(Path(filename).stem, bids_keys)
 
-    cwd = os.getcwd()
-    os.chdir(outdir)
+    # If bids_keys['task'] is still empty, stop the program
+    if not bids_keys['task']:
+        raise KeyError(f'No "task" attribute found')
 
-    heur = utils.load_heuristic(heur_file)
-    name = heur.heur(Path(filename).stem, name)
+    # Compose name by looping in the bids_keys dictionary
+    # and adding nonempty keys
+    name = ''
+    for key in bids_keys:
+        if bids_keys[key]:
```

Comment:
```suggestion
        if bids_keys[key] != '':
```

I know that `''` evaluates to `False` but it's good practice to only use the `if X` notation if `X` is expected to be a boolean.